apps/root/src/hooks/useStoredLabels.ts

```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import useLabelService from './useLabelService';
+import { AccountLabels } from '@types';
+import { isEqual } from 'lodash';
+
+function useStoredLabels() {
+  const labelService = useLabelService();
+  const [storedLabels, setStoredLabels] = React.useState<AccountLabels>(labelService.getStoredLabels());
+
+  const prevLabelsRef = React.useRef<AccountLabels>(storedLabels);
+
+  React.useEffect(() => {
+    const currentLabels = labelService.getStoredLabels();
+    if (!isEqual(prevLabelsRef.current, currentLabels)) {
+      setStoredLabels(currentLabels);
+      prevLabelsRef.current = currentLabels;
+    }
+  }, [labelService]);
```

Comment:
we should also add the accountService.getWallets value here, so it refreshes when the wallets/user changes

Reply:
Nice catch! Thank you